### PR TITLE
Add Python linter

### LINKED ISSUE #2033: Add Python linter and beautifier
So to check Python source code in this repo.

Fixing this issue doesn't check the Python generated by SQLFlow, which is tracked in https://github.com/sql-machine-learning/sqlflow/issues/2032

As a reference:

https://github.com/sql-machine-learning/elasticdl/blob/95c74b28c867dd5986724ad77d7802386624528d/.pre-commit-config.yaml#L1-L23

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fix https://github.com/sql-machine-learning/sqlflow/issues/2033 by ading pylint check to pre-commit.

This change introduce our own bash script scripts/pre-commit/pylint.sh that calls pylint only with changed files. Without this script, we would have to fix too many Python source files to make them pylint-compatible before we can merge this PR.

### CODE REVIEW

**Comment by tonyyang-svail** [comment]
@wangkuiyi, the linter check fails on some Python files, mostly code formatting and comment requirements. Since there is an ongoing code refactor by @typhoonzero at https://github.com/sql-machine-learning/sqlflow/pull/2026, how about we waiting for the merge of https://github.com/sql-machine-learning/sqlflow/pull/2026 then I will help to make sure all Python code satisfies the linter?

**Comment by wangkuiyi** [comment]
@tonyyang-svail Without the linter, I am afraid that refactorization work would introduce new incompatibilities. Let us find a way to decompose the work of adding lintering into smaller pieces.

**Comment by tonyyang-svail** [comment]
> Let us find a way to decompose the work of adding lintering into smaller pieces.

Sure.

**Review by tonyyang-svail — DISMISSED** [review]

**Comment by wangkuiyi** [comment]
@tonyyang-svail I reduced this change to add only pylint pre-commit hook, and only check changed files.

Since we already have yapf and isort, I don't think black and flake are must-to-have.

pylint doesn't accept `from some_pacakge import *`. So, if we change some Python file in the future, we will need to fix such imperfections.

File: scripts/docker/install-python.bash [hunk 1]
```diff
@@ -48,4 +46,7 @@ pip install \
     oss2==2.9.0 \
     plotille==3.7 \
     seaborn==0.9.0 \
-    jsbeautifier
+    jsbeautifier \
+    yapf \
+    isort \
```
Comment (tonyyang-svail) [inline scripts/docker/install-python.bash:51]: Do we need to follow the version number of the original code?

**Review by tonyyang-svail — APPROVED** [review]

File: scripts/docker/install-python.bash (hunk 1 above)
Comment (wangkuiyi) [reply scripts/docker/install-python.bash:51]: I had the same question. I feel that we can try sticking with the most recent version and see if their upgrading would make incompatible reformat suggestions.

**Pull request merged**